app/models/pacbio/library_factory.rb

```diff
@@ -57,7 +58,9 @@ def build_library(library_attributes)
 
     def create_library(library_attributes)
       library_attributes_without_requests = library_attributes.except(:requests)
-      Pacbio::Library.new(library_attributes_without_requests.merge!(tube: Tube.new))
+      library = Pacbio::Library.new(library_attributes_without_requests)
+      @container_material = ContainerMaterial.new(container: Tube.new, material: library)
```

Comment:
I think generally it's an anti-pattern to make things public just for the sakes of testing.  Unit tests should only test the public interface of a class, not the inner workings. 

This test is supposed to check that the state of the factory is correct if we haven't yet called save, which it is doing.  It would be nice to check that the ContainerMaterial was created, but it doesn't matter that it has been until you save.  We check the link exists in the context checking the state after the save.

I've not wanted to change the existing code base, but another anti-pattern is getting the initializer to do too much work.  Initialize would, in an ideal world, only save values for the class to use.  There should be a method called `build` which does the building of the factory's output.  Then you separate object creation from the object's purpose.